astropy/table/table.py

```diff
@@ -1244,18 +1245,30 @@ def __setitem__(self, item, value):
             name = item
             # If this is a column-like object that could be added directly to table
             if isinstance(value, BaseColumn) or self._add_as_mixin_column(value):
+                # If we're setting a new column to a scalar, broadcast it.
+                # (things will fail in _init_from_cols if this doesn't work)
+                if (len(self) > 1 and (getattr(value, 'isscalar', False) or
```

Comment:
I'm a little confused about `len(self) > 1`.  Was that intended?  We definitely should not consider a length=1 array to be the same as a scalar.